internal/venice-common/src/main/java/com/linkedin/venice/annotation/UnderDevelopment.java

```diff
@@ -0,0 +1,20 @@
+package com.linkedin.venice.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+
+/**
+ * Indicates that an API (class, method, or field) is under development and is not yet stable.
+ * This annotation warns users that the annotated element may change or be removed in future versions.
+ */
+@Retention(RetentionPolicy.CLASS) // Retained in class files but not available at runtime.
```

Comment:
I wonder if this is working though? If you mark the old API as under development as I comment above, then wouldn't unit test / integration test should all failed?

Reply:
I haven't added any compile-time checks for this, so this annotation is mainly informational.